tests/test_backends.py

```diff
@@ -587,3 +592,45 @@ def testAcrossVariantSets(self):
                         self.verifySearchVariants(
                             variantSets, referenceName, 0, 2 ** 32,
                             pageSize=pageSize)
+
+
+class TestCallSets(TestWormtableBackend):
+    """
+    Tests the searchCallSets end point
+    """
+    def testCallSetIds(self):
+        # verify that callSetIds are equal to sample columns in wormtable
+        for variantSetId in self.getVariantSetIds():
+            callSetIds = self.getCallSetIds(variantSetId)
+            variantSets = self._backend._variantSetIdMap[variantSetId]
+            sampleNames = set(variantSets._sampleNames)
+            self.assertEqual(callSetIds, sampleNames)
+
+    def testUniqueCallSetIds(self):
+        # verify that all callSetIds in each variantSet are unique
+        for variantSetId in self.getVariantSetIds():
+            callSetIds = set()
+            for callSetId in self.getCallSetIds(variantSetId):
+                self.assertTrue(callSetId not in callSetIds)
+                callSetIds.add(callSetId)
+
+    def testUniqueCallSetIdsAcrossVariantSets(self):
+        # verify that all callSetIds across different variantSets are unique
+        # if combining the variantSetIds with callSetIds
+        ids = set()
+        for variantSetId in self.getVariantSetIds():
+            for callSetId in self.getCallSetIds(variantSetId):
+                id = variantSetId + callSetId
+                self.assertTrue(id not in ids)
+                ids.add(id)
+
+    def testVariantSetIds(self):
+        # test the VariantSetId field of GACallSet object is correct
+        allCallSetIdMap = dict()
+        for variantSetId in self.getVariantSetIds():
+            for callSetId in self.getCallSetIds(variantSetId):
+                if callSetId not in allCallSetIdMap:
+                    allCallSetIdMap[callSetId] = []
+                allCallSetIdMap[callSetId].append(variantSetId)
+        wormtableCallSetIdMap = self._backend._callSetIdMap
```

Comment:
We're breaking isolation here: we should never access any variable starting with an `_` from outside its own class or one of its subclasses. Do we really need to build up a map of all the callSetIds in the system to check this in any case? Wouldn't it be simpler to get each callSet in turn and just verify that its variantSetId is correct?